cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -971,6 +971,21 @@ def make_resource(self, data, **kwargs):
         """Generate resource."""
         return BudgetSubscriber(**data)
 
+    @validates_schema
+    def checks_address_format(self, data, **kwargs):
+        """Check email/sns format using regex."""
+        subscription_type = data.get("subscription_type")
+        address = data.get("address")
+
+        if subscription_type == "EMAIL":
+            email_pattern = re.compile(r"[a-z0-9]+@[a-z]+\.[a-z]{2,3}")
```

Comment:
This regex also match name@gmial.comcomcom
Shall we all `^` at the beginning and `$` at the end?

Reply:
nit: the two lines code can be reduced to one line:
```
re.match(pattern, string)
```

Reply:
Good suggestion. Improved!